src/mlpack/methods/ann/make_alias.hpp

```diff
@@ -15,44 +15,128 @@
 #define MLPACK_METHODS_ANN_MAKE_ALIAS_HPP
 
 #include <mlpack/prereqs.hpp>
+#include <bandicoot>
 
 namespace mlpack {
 
+template<typename MatType>
+struct IsCootMatType;
+
+template<typename CubeType>
+struct IsCootCubeType;
+
+template<typename MatType>
+struct IsCootMatType
+{
+  constexpr static bool value = false;
+};
+
+template<typename CubeType>
+struct IsCootCubeType
+{
+  constexpr static bool value = false;
+};
+
+#ifdef MLPACK_HAS_COOT
+template<typename MatType>
+struct IsCootMatType
+{
+  constexpr static bool value = true;
+};
+
+template<typename CubeType>
+struct IsCootCubeType
+{
+  constexpr static bool value = true;
+};
+
+#endif
+
 /**
  * Reconstruct `m` as an alias around the memory `newMem`, with size `numRows` x
  * `numCols`.
  */
-template<typename MatType>
-void MakeAlias(MatType& m,
-               typename MatType::elem_type* newMem,
+template<typename InMatType,
+         typename OutMatType,
+         typename = typename std::enable_if<
+         !IsCootMatType<InMatType>::value>::type>
+void MakeAlias(OutMatType& m,
+               const InMatType& oldMat,
                const size_t numRows,
                const size_t numCols)
 {
   // We use placement new to reinitialize the object, since the copy and move
   // assignment operators in Armadillo will end up copying memory instead of
   // making an alias.
-  m.~MatType();
-  new (&m) MatType(newMem, numRows, numCols, false, true);
+  typename InMatType::elem_type* newMem = oldMat.memptr();
+  m.~Mat();
+  new (&m) OutMatType(newMem, numRows, numCols, false, true);
 }
 
 /**
  * Reconstruct `c` as an alias around the memory` newMem`, with size `numRows` x
  * `numCols` x `numSlices`.
  */
-template<typename CubeType>
-void MakeAlias(CubeType& c,
-               typename CubeType::elem_type* newMem,
+template<typename InCubeType,
+         typename OutCubeType,
+         typename = typename std::enable_if<
+         !IsCootCubeType<InCubeType>::value>::type>
+void MakeAlias(OutCubeType& c,
+               const InCubeType& oldCube,
                const size_t numRows,
                const size_t numCols,
                const size_t numSlices)
 {
   // We use placement new to reinitialize the object, since the copy and move
   // assignment operators in Armadillo will end up copying memory instead of
   // making an alias.
-  c.~CubeType();
-  new (&c) CubeType(newMem, numRows, numCols, numSlices, false, true);
+  typename InCubeType::elem_type* newMem = oldCube.memptr();
+  c.~Cube();
+  new (&c) OutCubeType(newMem, numRows, numCols, numSlices, false, true);
 }
 
+#ifdef MLPACK_HAS_COOT
```

Comment:
This also does not use the `coot::` namespace directly and so doesn't need to be guarded.